app/models/room.rb

```diff
@@ -16,7 +16,13 @@ class Room < ApplicationRecord
   validates :meeting_id, presence: true, uniqueness: true
 
   before_validation :set_friendly_id, :set_meeting_id, on: :create
+
   after_create :create_meeting_options
+  after_create :generate_viewer_access_code, if: :auto_generate_viewer_access_code?
+  after_create :generate_moderator_access_code, if: :auto_generate_moderator_access_code?
+
+  after_find :generate_viewer_access_code, if: :auto_generate_viewer_access_code?
+  after_find :generate_moderator_access_code, if: :auto_generate_moderator_access_code?
```

Comment:
I'm not sure how much I like this - this adds an extra query for a lot of our endpoints

Reply:
Yup, access codes aren't room properties now.
So, no matter what we'd have to make an extra query to fetch the room codes.
I haven't payed optimizing this logic any of my time,

We could create one callback concerned with the access codes auto generation, that callback will fetch both of the room codes with one query instead of having to make two with this current trivial implementation.
But, at least one query per room on each find will still be issued with the current design.

To eliminate the need for that we may consider falling back to access codes properties but we pay its design more time to reduce spatial (memory) wastage.

Reply:
This logic was optimized in #3762